src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/SatelliteDynamics.cpp

```diff
@@ -160,62 +165,47 @@ void SatelliteDynamics::DynamicalEquations(const Dynamics::StateVector& x, Dynam
         totalGravitationalAcceleration_SI += gravitationalAcceleration.inFrame(gcrfSPtr_, currentInstant).getValue();
     }
 
-    // Integrate position and velocity states
-    dxdt[0] = x[3];
-    dxdt[1] = x[4];
-    dxdt[2] = x[5];
-    dxdt[3] = totalGravitationalAcceleration_SI[0];
-    dxdt[4] = totalGravitationalAcceleration_SI[1];
-    dxdt[5] = totalGravitationalAcceleration_SI[2];
-}
+    dxdt[3] += totalGravitationalAcceleration_SI[0] ;
+    dxdt[4] += totalGravitationalAcceleration_SI[1] ;
+    dxdt[5] += totalGravitationalAcceleration_SI[2] ;
+
+    // DRAG
+    
+    const Real mass = satelliteSystem_.getMass().inKilograms() ;
+    const Real dragCoefficient = satelliteSystem_.getDragCoefficient() ;
+    const Real surfaceArea = satelliteSystem_.getCrossSectionalSurfaceArea() ;
+
+    for (const auto& objectName : environment_.getObjectNames())
+    {
+
+        // TBI: currently only defined for Earth
+        if (environment_.accessCelestialObjectWithName(objectName)->accessAtmosphericModel() != nullptr)
+        {
+
+            const Real rho = environment_.accessCelestialObjectWithName(objectName)->getAtmosphericDensityAt(currentPosition).getValue() ;
```

Comment:
Maybe we should be explicit and call this atmospheric density since we are using the full terms for everything else.
```suggestion
            const Real atmosphericDensity = environment_.accessCelestialObjectWithName(objectName)->getAtmosphericDensityAt(currentPosition).getValue() ;
```